Fix Mold Breaker Knock Off with Sticky Hold

ascriptmaster opened the pull request:
Knock Off's function call for target.takeItem was written in such a way
that Sticky Hold could not actively notice and trigger its TakeItem event.
This fixes the function to have the handler called properly.

ascriptmaster commented:
@Marty-D, can you verify behavior of Gen5 and below Pokemon fainting during Knock Off because of Rough Skin or Rocky Helmet? And regardless of your answer, merge this so it can be patched, and if Gen5 Knock Off is wrong I'll PR it later since it isn't very high priority.

Marty-D commented:
Gen 5 and 6 are the same for this stuff. I think the reason Knock Off was written that way was because of Mail. Check if that still works in Gen 5, and if so I'll merge when I get home in about 4 hours.

Also, the Plate stealing thing is based on Arceus in Gen 5 and 6, not Multitype (not that it matters for your test).

ascriptmaster commented:
Mail has been working ever since I fixed it all the way back in e27e7de08a0dd662e31aa3583fac7a76e3ca20a4, and there are no implementation changes through the generations so it's going to work in Gen 5. Regardless, I fixed the Knock Off test to clarify that it checks Arceus.

Marty-D commented:
Oh wow, I didn't even remember that even though I code reviewed it at the time. :+1:
